packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -82,6 +82,67 @@ export const commandIDs = {
   addFileToPipeline: 'pipeline-editor:add-node'
 };
 
+const createPalette = (categories: any[]): any => {
+  const palette = {
+    version: '3.0' as '3.0',
+    categories: categories ?? []
+  };
+
+  for (const category of categories) {
```

Comment:
We shouldn't do this on the front-end, all info required should come from the services. @kiersten-stokes

Reply:
Yeah, if we need any changes to the palette that is being returned from the call to `/elyra/pipeline/components/[runtime]/[component-id]/properties`, I'm happy to make them!

Reply:
@lresende we have to piece the palette together at the moment because the `pipeline-editor` expects the palette to have all the properties embedded.

We originally did this because the pipeline needs all properties to validate. However, I want to change this to be a bit more async so that we can lazily fetch properties for validation